codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -165,9 +162,14 @@ public void execute() throws MojoExecutionException {
                     logger.info("Generating java file for YAML file [" + dataFile.getName() + "] in domain ["
                             + domain + "]");
                     
-                    String tempPackage = basePackage;
+                    String tempPackage = "";
+                    // add the basePackage, iff it contains a value
+                    if (!basePackage.isEmpty()) {
+                        tempPackage += basePackage;
+                    }
+                    // add the folder, iff it contains a value
                     if (!folder.isEmpty()) {
-                        tempPackage += "." + folder;
+                        tempPackage += tempPackage.isEmpty() ?  folder :  "." + folder;
                     }
                     helper.generateNewCode(dataFile, relativePath, tempPackage, domain);
```

Comment:
so how are we handling an empty base package now?  Do a user get a valid error on empty package?